plugin/storage/integration/es_index_cleaner_test.go

```diff
@@ -216,3 +299,28 @@ func createESClient() (*elastic.Client, error) {
 		elastic.SetURL(queryURL),
 		elastic.SetSniff(false))
 }
+
+func getVersion(client *elastic.Client) (uint, error) {
+	pingResult, _, err := client.Ping(queryURL).Do(context.Background())
+	if err != nil {
+		return 0, err
+	}
+	esVersion, err := strconv.Atoi(string(pingResult.Version.Number[0]))
+	if err != nil {
+		return 0, err
+	}
+	return uint(esVersion), nil
+}
+
+func createILMPolicy(client *elastic.Client, policyName string) error {
+	_, err := client.XPackIlmPutLifecycle().Policy(policyName).BodyString("{\"policy\": {\"phases\": {\"hot\": {\"min_age\": \"0ms\",\"actions\": {\"rollover\": {\"max_age\": \"1d\"},\"set_priority\": {\"priority\": 100}}}}}}").Do(context.Background())
+	return err
+}
+
+func cleanES(t *testing.T, client *elastic.Client, policyName string) {
+	_, err := client.DeleteIndex("*").Do(context.Background())
+	require.NoError(t, err)
+	_, err = client.XPackIlmDeleteLifecycle().Policy(policyName).Do(context.Background())
```

Comment:
`require.NoError(t, err)` for this as well.

Reply:
Unfortunately XPackILMDeleteLifecycle() returns 404 - if the ILM is not present. May be I have to do the check myself and delete if present; will do and add Noerror check

Reply:
Added a check to fail if err != nil && is not 404.

https://github.com/jaegertracing/jaeger/pull/2454/commits/8a3cf15af0ec9d916a9e3a4f9a3be0e221f00211